### PR TITLE
PermissionScope clarifications

### PR DESCRIPTION
**Pull request by jglick**
Some aspects of `PermissionScope` were muddled.

@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/model/Computer.java [hunk 1]
```diff
@@ -1701,7 +1701,7 @@ public long getWhen() {
      */
     public static final Permission EXTENDED_READ = new Permission(PERMISSIONS,"ExtendedRead", Messages._Computer_ExtendedReadPermission_Description(), CONFIGURE, Boolean.getBoolean("hudson.security.ExtendedReadPermission"), new PermissionScope[]{PermissionScope.COMPUTER});
     public static final Permission DELETE = new Permission(PERMISSIONS,"Delete", Messages._Computer_DeletePermission_Description(), Permission.DELETE, PermissionScope.COMPUTER);
-    public static final Permission CREATE = new Permission(PERMISSIONS,"Create", Messages._Computer_CreatePermission_Description(), Permission.CREATE, PermissionScope.COMPUTER);
+    public static final Permission CREATE = new Permission(PERMISSIONS,"Create", Messages._Computer_CreatePermission_Description(), Permission.CREATE, PermissionScope.JENKINS);
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/Computer.java]: UHH, is it safe?
At least Role Strategy has a hack, which allows to allow creating jobs with a specific pattern only. Such change may impact it's UIs (I need to check it)

File: core/src/main/java/hudson/security/PermissionScope.java [hunk 2]
```diff
@@ -56,7 +58,7 @@
     /**
      * Domain model type that approximates this scope.
      */
-    public final Class modelClass;
+    public final Class<? extends ModelObject> modelClass;
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/security/PermissionScope.java]: AFAIK it's not binary compatible :(

File: core/src/main/java/hudson/security/PermissionScope.java [hunk 3]
```diff
@@ -65,7 +67,7 @@
      */
     private final Set<PermissionScope> containers;
 
-    public PermissionScope(Class modelClass, PermissionScope... containers) {
+    public PermissionScope(Class<? extends ModelObject> modelClass, PermissionScope... containers) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/security/PermissionScope.java]: same

**Comment by oleg-nenashev** [comment]
:bug: for binary compatibility issue. Needs a fix IMHO.

File: core/src/main/java/hudson/model/Computer.java (hunk 1 above)
Comment (jglick) [inline core/src/main/java/hudson/model/Computer.java]: This is `Computer.CREATE` not `Item.CREATE`.

File: core/src/main/java/hudson/security/PermissionScope.java (hunk 2 above)
Comment (jglick) [inline core/src/main/java/hudson/security/PermissionScope.java]: It is binary compatible, just not source compatible (without `unchecked` warnings anyway)

**Comment by olivergondza** [comment]
:+1:

**Comment by daniel-beck** [comment]
:+1:

**Comment by oleg-nenashev** [comment]
Agreed on binary compatibility. :bee:

**Pull request merged**